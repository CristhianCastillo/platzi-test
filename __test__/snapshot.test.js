import { getCharacter } from '../snapshot';
import rick from '../rick.json';

describe('Es hora de las instantaneas', () => {
    test('Shapshot', () => {
        expect(getCharacter(rick)).toMatchSnapshot();
    });

    test('Siempre fallara las instantaneas', () => {
        const user = {
            createAt: new Date(),
            id: Math.floor(Math.random() * 20),
        };
        expect(user).toMatchSnapshot();
    });

    test('Tenemos una exception dentro del snapshot', () => {
        const user = {
            id: Math.floor(Math.random() * 20),
            name: 'Rick'
        };
        expect(user).toMatchSnapshot({
            id: expect.any(Number)
        });
    });
});