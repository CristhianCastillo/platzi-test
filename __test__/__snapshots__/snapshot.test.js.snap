// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Es hora de las instantaneas Shapshot 1`] = `
Object {
  "gender": "Male",
  "id": 1,
  "name": "Rick Sanches",
  "species": "Human",
  "status": "Alive",
}
`;

exports[`Es hora de las instantaneas Siempre fallara las instantaneas 1`] = `
Object {
  "createAt": 2021-09-09T02:50:03.514Z,
  "id": 6,
}
`;

exports[`Es hora de las instantaneas Tenemos una exception dentro del snapshot 1`] = `
Object {
  "id": Any<Number>,
  "name": "Rick",
}
`;
